### PR TITLE
Make sorting respect nested tasks

### PR DESCRIPTION
**Pull request by vishar0**
Fix for https://github.com/usernamehw/vscode-todo-md/issues/25

**Tested with the following example:**
```
(C) task 1
  (B) subtask 1a
  (A) subtask 1b
(C) task 2 {due:2023-08-30}
  (B) subtask 2a {due:2023-08-20}
  subtask 2b {due:2023-08-10}
  (A) subtask 2c
(A) task 3
(A) task 4 {due:2023-08-05}
  (B) subtask 4a
  (C) subtask 4b
```

Default sort output **before** with nesting broken:
```
(A) task 4 {due:2023-08-05}
  subtask 2b {due:2023-08-10}
  (B) subtask 2a {due:2023-08-20}
(C) task 2 {due:2023-08-30}
  (A) subtask 1b
  (A) subtask 2c
(A) task 3
  (B) subtask 1a
  (B) subtask 4a
(C) task 1
  (C) subtask 4b
```

Defaults sort output **after** with tree structure maintained and subtasks sorted among themselves:
```
(A) task 4 {due:2023-08-05}
  (B) subtask 4a
  (C) subtask 4b
(C) task 2 {due:2023-08-30}
  subtask 2b {due:2023-08-10}
  (B) subtask 2a {due:2023-08-20}
  (A) subtask 2c
(A) task 3
(C) task 1
  (A) subtask 1b
  (B) subtask 1a
```

### CODE REVIEW

**Comment by viswanathgs** [comment]
@usernamehw, any thoughts on this? Thanks in advance!

File: src/commands.ts [hunk 1]
```diff
@@ -212,17 +212,31 @@ export function sortTasksInEditor(editor: TextEditor, edit: TextEditorEdit, sort
...
 		lineEnd = editor.document.lineCount - 1;
 	}
+
+	// Fetch only the top-level tasks within the selection to avoid messing up
+	// nested tasks after sorting and to retain the tree structure.
 	const tasks: TheTask[] = [];
-	for (let i = lineStart; i <= lineEnd; i++) {
-		const task = getTaskAtLineExtension(i);
+	let lineno = lineStart;
+	while (lineno <= lineEnd) {
```
Comment (usernamehw) [inline src/commands.ts:220]: Could you use `for` loop instead of `while` everywhere?

**Comment by usernamehw** [comment]
Doesn't work for incomplete selections:

```
(C) parent
    (A) nested 1
        (C) nested 1.1
        (D) nested 1.2
    (B) nested 2
```

Select lines 3-5 (C) (D) (B) => sort by priority => (C) (D) now have wrong parent task.

**Comment by viswanathgs** [comment]
> Doesn't work for incomplete selections:

Should be an easy fix - I currently handle the case where there is incomplete selection in the "end" but not in the "start".

But, what would be the right thing here? If L3-5 is selected, do you want to skip L3-4 (the incomplete part) or extend the selection to L2-5 to make it complete?

**Comment by usernamehw** [comment]
Maybe a noop? With a warning notification to the user.

**Comment by viswanathgs** [comment]
Stale, will attempt to cleanup

**Pull request closed without merging**